### PR TITLE
Add Good Customers Boxes

### PR DESCRIPTION
**Pull request by defrance**

### CODE REVIEW

**Comment by eldy** [comment]
To avoid introducing french key now, can you rename
bonclient into  "payingcustomer" for example ?
Also what is your definition of good customer ? When it is good or not.
Do you mean "have paid 100% of invoice" ?

**Comment by defrance** [comment]
"Good customers" is for me "recurent customers" with all it's bills payed
that why i order by number of bill and display the number of unpayed bills in brackets

**Comment by aljawaid** [comment]
Nice idea. Could be useful if we could manually choose/select good/loyal or vip customers too.

File: htdocs/core/boxes/box_goodcustomers.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        include_once DOL_DOCUMENT_ROOT.'/societe/class/societe.class.php';
+        $thirdpartystatic=new Societe($db);
+
+        $this->info_box_head = array('text' => $langs->trans("BoxTitleGoodCustomers",$max));
+
+		if ($user->rights->societe->lire)
+		{
+			
+			$sql = "SELECT s.rowid, s.nom as name, s.logo, s.code_client, s.code_fournisseur, s.client, s.fournisseur, s.tms as datem, s.status as status,";
+			$sql.= " count(*) as nbfact, sum(if (f.paye=1,1,0)) as nbfactpaye";
```
Comment (eldy) [inline htdocs/core/boxes/box_goodcustomers.php]: Can you replace the hardcoded if with the portable   $db->ifsql  function

Comment (defrance) [inline htdocs/core/boxes/box_goodcustomers.php]: that's made

**Pull request merged**